src/backend/marsha/core/views.py

```diff
@@ -397,6 +408,183 @@ class DocumentView(BaseLTIView):
+        )
+
+
+@method_decorator(csrf_exempt, name="dispatch")
+@method_decorator(xframe_options_exempt, name="dispatch")
+class LTISelectView(TemplateResponseMixin, View):
+    """LTI view called to select LTI content through Deep Linking.
+
+    It is designed to work as a React single page application.
+
+    """
+
+    template_name = "core/resource.html"
+
+    def get_context_data(self):
+        """Build context for template rendering of configuration data for the frontend.
+
+        Returns
+        -------
+        dictionary
+            context with configuration data for the frontend
+
+        """
+
+        def _manage_exception(error):
+            logger.warning(str(error))
+            return {"state": "error"}
+
+        lti = LTI(self.request)
+
+        if not lti.is_instructor and not lti.is_admin:
+            raise PermissionDenied
+
+        try:
+            lti.verify()
+            Playlist.objects.get_or_create(
+                lti_id=lti.context_id,
+                consumer_site=lti.get_consumer_site(),
+                defaults={"title": lti.context_title},
+            )
+        except LTIException as error:
+            app_data = _manage_exception(error)
+        else:
+            app_data = self._get_app_data(lti)
+
+        return {
+            "app_data": json.dumps(app_data),
+            "static_base_url": f"{settings.STATIC_URL}js/",
+            "external_javascript_scripts": settings.EXTERNAL_JAVASCRIPT_SCRIPTS,
+        }
+
+    def _get_app_data(self, lti):
+        """Build app data for the frontend with information retrieved from the LTI launch request.
+
+        Returns
+        -------
+        dictionary
+            Configuration data to bootstrap the frontend:
+            A form is build in react to post data to the LTI Consumer thnough an iframe.
+
+            For instructors only
+            ++++++++++++++++++++
+
+            - lti_select_form_action_url: URL to post data to.
+            - lti_select_form_data: Data to post.
+            - new_document_url: LTI URL to a new document.
+            - new_video_url: LTI URL to a new video.
+            - documents: Documents list with theirs LTI URLs.
+            - videos: Videos list with theirs LTI URLs.
+
+        """
+        documents = DocumentSelectLTISerializer(
+            get_selectable_resources(Document, lti),
+            many=True,
+            context={"request": self.request},
+        ).data
+
+        videos = VideoSelectLTISerializer(
+            get_selectable_resources(Video, lti),
+            many=True,
+            context={"request": self.request},
+        ).data
+
+        new_uuid = str(uuid.uuid4())
+        app_data = _get_base_app_data()
+        lti_select_form_data = {
+            "lti_response_url": self.request.POST.get("content_item_return_url")
+        }
+        lti_select_form_data.update(self.request.POST)
+        lti_select_form_data.update({"lti_message_type": "ContentItemSelection"})
```

Comment:
I'm not confortable with these 3 lines. Copying the POST dict and modifying it would seem more straightforward. Also is there a way to use the same name for "content_item_return_url" and "lti_response_url" or are these names imposed by LTI?

Reply:
content_item_return_url only is in the specs. I'll replace lti_response_url